### ISSUE TITLE
label-mail is useless to me because go mail header parsing is weak

### ISSUE THREAD

**Issue by zostay** [zostay/dotfiles-go#1]
I'm waiting for this to be resolved

https://github.com/emersion/go-message/issues/110

I could monkey patch that or do some sort of catch/release system on extra long headers, but there's active work on this issue, so I'm hoping it will be resolved soon. I will continue to use my old label-mail program until that's fixed or I become impatient again.

**Comment by zostay** [comment]
The problem in go-message has been corrected. I have tried out my mail with code on master. The behavior previously seen was that any header longer than 4000 bytes would silently cause the rest of the message to cease being parsed. This is no longer the case with latest go-message master.

However, I now have a new problem. The RFC 5322 date parsing in Go's built-ins is unacceptably unforgiving. In particularly, it does not implement the obsolete time zone handling of that RFC at all. I need to rewrite my date handling to intercept the ambiguous and awful error from `ParseDate()`, "mail: header could not be parsed", and add a more forgiving fallback date parser in those cases.
